client/my-sites/draft/index.jsx

```diff
@@ -152,6 +152,11 @@ module.exports = React.createClass( {
 			imageUrl = '//' + image.hostname + image.pathname + '?w=680px';
 		}
 
+		if ( post && post.canonical_image ) {
+			image = url.parse( post.canonical_image.uri, true );
+			imageUrl = '//' + image.hostname + image.pathname + '?w=680px';
```

Comment:
Won't this always try to load the image from HTTPS when viewed on WordPress.com ? What if the image is from a self-hosted Jetpack site without Photon enabled? Also, for the resizing parameter, shouldn't we use [`lib/resize-image-url`](https://github.com/Automattic/wp-calypso/blob/master/client/lib/resize-image-url/index.js)?